test-data/unit/check-functions.test

```diff
@@ -1479,7 +1479,7 @@ L = Callable[[Arg(name='x', type=int)], int] # ok
 # I have commented out the following test because I don't know how to expect the "defined here" note part of the error.
 # M = Callable[[Arg(gnome='x', type=int)], int]   E: Invalid type alias   E: Unexpected keyword argument "gnome" for "Arg"
 N = Callable[[Arg(name=None, type=int)], int] # ok
-O = Callable[[List[Arg(int)]], int]  # E: Invalid type alias  # E: Value of type "int" is not indexable  # E: Type expected within [...] # E: The type List[T] is not generic and not indexable
+O = Callable[[List[Arg(int)]], int]  # E: Invalid type alias  # E: Value of type "int" is not indexable  # E: Type expected within [...] # E: The type Type[List[T]] is not generic and not indexable
```

Comment:
> E: The type ``Type[List[T]]`` is not generic and not indexable.

@sixolet This error looks quite confusing in this context, since the type mentioned is generic and is indexable. In real mypy installation (not in tests) I get these errors:
```
tst26.py:14: error: Invalid type alias
tst26.py:14: error: Value of type "object" is not indexable
tst26.py:14: error: Type expected within [...]
tst26.py:14: error: Value of type List[_T] is not indexable
```
Note that _none_ of them mentions (or even hints) that something is wrong with ``Arg``. Is there an issue for improving the error message in such cases?